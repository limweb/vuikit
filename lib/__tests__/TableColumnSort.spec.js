import { createVue, destroyVM, queryByTag, toJSON } from './_util'
import waitForUpdate from './_wait-for-update'

const testDataArr = []
const getTestData = function () {
  return [
    { name: 'Toy Story', release: '1995-11-22', director: 'John Lasseter', runtime: 80 },
    { name: 'A Bug\'s Life', release: '1998-11-25', director: 'John Lasseter', runtime: 95 },
    { name: 'Toy Story 2', release: '1999-11-24', director: 'John Lasseter', runtime: 92 },
    { name: 'Monsters, Inc.', release: '2001-11-2', director: 'Peter Docter', runtime: 92 },
    { name: 'Finding Nemo', release: '2003-5-30', director: 'Andrew Stanton', runtime: 100 }
  ]
}

getTestData().forEach(cur => {
  Object.keys(cur).forEach(prop => {
    testDataArr.push(cur[prop].toString())
  })
})

describe.skip('Table Column Sort', () => {
  let vm
  afterEach(() => {
    destroyVM(vm)
  })

  it('renders correctly', done => {
    vm = createVue({
      template: `
        <vk-table :data="testData">
          <vk-table-column-sort headerClass="myHeaderClass" cellClass="myCellClass">
          <vk-table-column-sort>
            <template slot="header" scope="props">
              Custom Header
            </template>
            <template slot="cell" scope="props">
              Custom Cell
            </template>
          </vk-table-column-sort>
        </vk-table>
      `,
      data: () => ({
        testData: getTestData()
      })
    })
    waitForUpdate(() => {
      expect(toJSON(vm)).toMatchSnapshot()
    }).then(done)
  })

  it('triggers event sort when the column header is clicked', done => {
    const vm = createVue({
      template: `
        <vk-table :data="testData">
          <vk-table-column-sort />
        </vk-table>
      `,
      data: () => ({
        testData: getTestData()
      })
    }, true)
    const table = queryByTag(vm, 'vk-table')
    const cb = jest.fn()

    table.$on('sort', cb)
    waitForUpdate(() => {
      vm.$el.querySelector('thead th').click()
    }).then(() => {
      const cbArgs = cb.mock.calls[0]
      expect(cb).toHaveBeenCalled()
      expect(cbArgs).toHaveLength(1)
    }).then(done)
  })
})
