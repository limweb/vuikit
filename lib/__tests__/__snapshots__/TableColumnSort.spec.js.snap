exports[`Table Column Sort renders correctly 1`] = `
Object {
  "child": Array [
    Object {
      "attr": Object {
        "class": "uk-table",
      },
      "child": Array [
        Object {
          "node": "element",
          "tag": "col",
        },
        Object {
          "child": Array [
            Object {
              "child": Array [
                Object {
                  "attr": Object {
                    "class": Array [
                      "myHeaderClass",
                      "uk-visible-hover-inline",
                      "vk-table-order",
                    ],
                  },
                  "child": Array [
                    Object {
                      "attr": Object {
                        "class": "uk-position-relative",
                      },
                      "child": Array [
                        Object {
                          "attr": Object {
                            "class": Array [
                              "uk-icon-justify",
                              "uk-margin-small-left",
                              "uk-invisible",
                              "vk-icon-arrow-down",
                            ],
                          },
                          "node": "element",
                          "tag": "i",
                        },
                      ],
                      "node": "element",
                      "tag": "span",
                    },
                  ],
                  "node": "element",
                  "tag": "th",
                },
              ],
              "node": "element",
              "tag": "tr",
            },
          ],
          "node": "element",
          "tag": "thead",
        },
        Object {
          "child": Array [
            Object {
              "child": Array [
                Object {
                  "attr": Object {
                    "class": "myCellClass",
                  },
                  "node": "element",
                  "tag": "td",
                },
              ],
              "node": "element",
              "tag": "tr",
            },
            Object {
              "child": Array [
                Object {
                  "attr": Object {
                    "class": "myCellClass",
                  },
                  "node": "element",
                  "tag": "td",
                },
              ],
              "node": "element",
              "tag": "tr",
            },
            Object {
              "child": Array [
                Object {
                  "attr": Object {
                    "class": "myCellClass",
                  },
                  "node": "element",
                  "tag": "td",
                },
              ],
              "node": "element",
              "tag": "tr",
            },
            Object {
              "child": Array [
                Object {
                  "attr": Object {
                    "class": "myCellClass",
                  },
                  "node": "element",
                  "tag": "td",
                },
              ],
              "node": "element",
              "tag": "tr",
            },
            Object {
              "child": Array [
                Object {
                  "attr": Object {
                    "class": "myCellClass",
                  },
                  "node": "element",
                  "tag": "td",
                },
              ],
              "node": "element",
              "tag": "tr",
            },
          ],
          "node": "element",
          "tag": "tbody",
        },
      ],
      "node": "element",
      "tag": "table",
    },
  ],
  "node": "root",
}
`;
